### PR TITLE
feat: update README

### PR DESCRIPTION
**Pull request by syedsajjadkazmii**
This PR updates README and adds PR template as contribution guideline

[VAN-1164](https://2u-internal.atlassian.net/browse/VAN-1164)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## Merge Checklist
+
+* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
+* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
+* [ ] Were your changes tested in the app?
+* [ ] Is there adequate test coverage for your changes?
+
+## Post-merge Checklist
+
+* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.
```
Comment (zainab-amir) [inline .github/pull_request_template.md:18]: Add another item
```suggestion
* [ ] Deploy the changes to prod after verifying on stage
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+## Description
+
+Include a description of your changes here, along with a link to any relevant Jira tickets and/or Github issues.
+
+### Deploy Preview
+
+Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).
```
[the remark this answers is not in the document]
Comment (zainab-amir) [reply .github/pull_request_template.md:7]: Add another section for tests.

**Review by zainab-amir — COMMENTED** [review]
Let's review this change over call because I think we can improve the PR template.

**Comment by syedsajjadkazmii** [comment]
> Let's review this change over call because I think we can improve the PR template.

@zainab-amir  thanks for the review. Could you please check now.

File: README.rst [hunk 3]
```diff
@@ -1,48 +1,151 @@
...
+
+- Register page
+
+- Login page
+
+- Forgot password page
+
+- Reset password page
+
+- Welcome page
```
Comment (zainab-amir) [inline README.rst:23]: Remove this.

File: README.rst [hunk 4]
```diff
@@ -1,48 +1,151 @@
...
+   * - ``INFO_EMAIL``
+     - The valid email address for information query regarding the target environment.
+     - ``info@example.com``
+
+   * - ``ENABLE_DYNAMIC_REGISTRATION_FIELDS``
+     - Enables support for dynamic registration fields that helps in adding more fields in register form for getting more details about users.
+     - ``true`` | ``''`` (empty strings are falsy)
+
+   * - ``DISABLE_ENTERPRISE_LOGIN``
+     - Enables support for disabling enterprise login.
```
Comment (zainab-amir) [inline README.rst:102]: nit:
```suggestion
     - Disables the enterprise login from Authn MFE.
```

File: README.rst [hunk 5]
```diff
@@ -1,48 +1,151 @@
...
+   * - ``PRIVACY_POLICY``
+     - The fully-qualified URL to the Privacy policy page in the target environment.
+     - ``https://example.com/privacy``
+
+   * - ``INFO_EMAIL``
+     - The valid email address for information query regarding the target environment.
+     - ``info@example.com``
+
+   * - ``ENABLE_DYNAMIC_REGISTRATION_FIELDS``
+     - Enables support for dynamic registration fields that helps in adding more fields in register form for getting more details about users.
```
Comment (zainab-amir) [inline README.rst:98]:
```suggestion
     - Enables support for configurable registration fields on the MFE. This flag must be enabled to show any registration field besides the default fields (`name`, `email`, `username`, `password`).
```

File: README.rst [hunk 6]
```diff
@@ -1,48 +1,151 @@
...
+.. list-table:: edX-specific Environment Variables
+   :widths: 30 50 20
+   :header-rows: 1
+
+   * - Name
+     - Description / Usage
+     - Example
+
+   * - ``ENABLE_PROGRESSIVE_PROFILING``
+     - Enables support for progressive profiling that helps in redirecting to welcome page and getting more details about users.
```
Comment (zainab-amir) [inline README.rst:119]:
```suggestion
     - Enables support for progressive profiling. If enabled, users are redirected to a second page where data for optional registration fields can be collected. 
```

Comment (zainab-amir) [reply README.rst:119]: Now that I think about it, I think this should be moved to the above table. The progressive profiling page is needed for the optional fields and should be available to openedX users as well.

**Review by zainab-amir — APPROVED** [review]

**Pull request merged**